### PR TITLE
backup: support backup with rate limit

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

Support backup with rate limit, so we can limit the impact to online query.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

Unit tests.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No,

### CODE REVIEW

File: components/backup/src/writer.rs [hunk 1]
```diff
@@ -79,34 +83,36 @@ impl BackupWriter {
...
-            let name = format!("{}_{}.sst", name, cf);
-            let checksum = tikv_util::file::calc_crc32_bytes(contents);
-            storage.write(&name, &mut contents as &mut dyn std::io::Read)?;
-            let mut file = File::new();
-            file.set_crc32(checksum);
-            file.set_name(name);
-            Ok(file)
-        };
+        let save_and_build_file =
+            |cf, mut contents: &[u8], limiter: Option<&Arc<IOLimiter>>| -> Result<File> {
```
Comment (kennytm) [inline components/backup/src/writer.rs]: Since the `limiter` is always `.cloned()` why not just pass `Option<Arc<IOLimiter>>` here?

**Comment by NingLin-P** [comment]
CI failed

**Comment by overvenus** [comment]
Caused by an unstable test test_server_pd_transfer_leader.

**Pull request merged**